### PR TITLE
Expose jwt-secret flag in erigon

### PR DESCRIPTION
**Pull request by enriavil1**

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -329,6 +329,12 @@ var (
 		Value: node.DefaultEngineHTTPPort,
 	}
 
+	JWTSecretPath = cli.StringFlag{
+		Name:  "jwt-secret",
+		Usage: "Token to ensure safe connection between CL and EL",
+		Value: "",
```
Comment (mandrigin) [inline cmd/utils/flags.go:335]: Do we want the default to be empty?

File: turbo/cli/flags.go [hunk 2]
```diff
@@ -157,6 +157,12 @@ var (
 		Name:  "healthcheck",
 		Usage: "Enable grpc health check",
 	}
+
+	JWTSecretPath = cli.StringFlag{
```
Comment (mandrigin) [inline turbo/cli/flags.go:161]: Why is it defined in 2 places?

Comment (enriavil1) [reply turbo/cli/flags.go]: I deleted this one. I will use the one inside of utils for consistency.

File: cmd/utils/flags.go (hunk 1 above)
Comment (enriavil1) [reply cmd/utils/flags.go:335]: If value is empty then it will create a jwt secret for us.

Comment (enriavil1) [reply cmd/utils/flags.go:335]: The default path is this: `const JwtDefaultFile = "jwt.hex"`

Comment (mandrigin) [reply cmd/utils/flags.go:335]: I would assume that if a file doesn’t exist, it will be created, regardless of a value here.

Comment (enriavil1) [reply cmd/utils/flags.go:335]: Currently it returns error if the file that we are trying to access doesn't exist. I am thinking of
doing a PR and changing two things now on the way we obtain JWT. 

1. If file doesn't exist then use default.

2. If file exists, but contains no jwt token then generate one for that file.

Comment (mandrigin) [reply cmd/utils/flags.go:335]: (1) is very non obvious, will be very difficult to debug subtle issues

Comment (enriavil1) [reply cmd/utils/flags.go:335]: Sorry, I was mistaken 1 is already one

Comment (mandrigin) [reply cmd/utils/flags.go:335]: I think the correct behaviour that I would expect would be.

1. There is a default value of this flag, which is `jwt.hex` in the current dir.
2. I can override it.
3. If this file exists -- try to read it, if it isn't a valid secret -- we error out.
4. If this file doesn't exist -- we create it.

Part 4 is the most questionable.

Comment (enriavil1) [reply cmd/utils/flags.go:335]: That is the current behavior.I talked to @Giulio2002 and cleared up my question. I thought that we would have to create a token if the file we pick doesn't have one, but that's not the case. The behavior is exactly how you described it.

Comment (mandrigin) [reply cmd/utils/flags.go:335]: then just pre-fill this value with `jwt.hex` (but use the existing constant) and that's it :)

File: cmd/rpcdaemon/cli/config.go [hunk 3]
```diff
@@ -564,7 +564,7 @@ func isWebsocket(r *http.Request) bool {
 // a new secret and stores to the default location.
 func obtainJWTSecret(cfg httpcfg.HttpCfg) ([]byte, error) {
 	var fileName string
-	if len(cfg.JWTSecretPath) > 0 {
+	if cfg.JWTSecretPath != JwtDefaultFile {
```
Comment (mandrigin) [inline cmd/rpcdaemon/cli/config.go:567]: This `if` is pointless now. If you just keep the line 572 it will be the same logic with less code.

**Review by mandrigin — CHANGES_REQUESTED** [review]

**Comment by Giulio2002** [comment]
can you also rename the flag id to `--authrpc.jwtsecret`, so that we are compatible with go-ethereum

File: cmd/rpcdaemon/cli/config.go (hunk 3 above)
Comment (enriavil1) [reply cmd/rpcdaemon/cli/config.go]: I got ride of the filename var since it is useless now too

**Review by mandrigin — APPROVED** [review]
👍

**Pull request merged**